lib/controller/authentication_controller.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:firebase_auth/firebase_auth.dart';
+
+class AuthenticationController {
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+
+  Future<void> signUp(String email, String password) async {
+    try {
+      await _auth.createUserWithEmailAndPassword(
+          email: email, password: password);
+    } catch (e) {
+      rethrow;
+    }
+  }
```

Comment:
Remove the try-catch blocks from each method.

```suggestion
Future<void> signUp(String email, String password) => _auth.createUserWithEmailAndPassword(email: email, password: password);
```

Reply:
why?
what if there is an error?